src/simulations/action_translator.py

```diff
@@ -707,6 +739,28 @@ def _flex_store_actions(self, res, time_step):
 
         return store_actions, store_bool_flex
 
+    def _flex_q_car_actions(self, res, time_step):
+        """Compute the flexible battery reactive power action from the optimisation result."""
+        no_flex_actions = self._get_no_flex_actions('flexible_q_car_action')
+        flexible_q_car_actions = np.zeros(self.n_homes)
+        for home in range(self.n_homes):
+            active_power = res['charge'][home, time_step] - res['discharge_other'][home, time_step]
+            max_q_car_flexibility = np.sqrt(self.max_apparent_power_car**2 - active_power**2)
+            flexible_q_car_actions[home] = res['q_car_flex'][home, time_step] / max_q_car_flexibility
+        # if action is close to zero, consider it to be zero
+        flexible_q_car_actions[home] = \
+            0 if -1e-3 <= res['q_car_flex'][home, time_step] <= 1e-3 else flexible_q_car_actions[home]
+        q_car_bool_flex = \
```

Comment:
the `bool_flex` booleans are not about whether there was charge or discharge, but about whether there was flexibility to make a choice or not. So here it would be about whether `max_q_car_flexibility` is nonzero.

Reply:
`q_car_flex` could be zero but the agent had a choice and set it as zero. In which case this is a meaningful action.